File: src/main/java/com/authenhub/service/PasswordManagerService.java
package com.authenhub.service;

import com.authenhub.bean.common.ApiResponse;
import com.authenhub.bean.passwordmanager.*;
import com.authenhub.utils.PasswordUtils;

import java.util.List;

public interface PasswordManagerService {
    /**
     * Search for password entries based on criteria
     *
     * @param request search criteria
     * @return API response with search results
     */
    ApiResponse<?> searchPassword(PwManagerSearchReq request);

    /**
     * Get a password entry by ID
     *
     * @param id password entry ID
     * @param includePassword whether to include the decrypted password
     * @return API response with the password entry
     */
    ApiResponse<PasswordResponse> getPasswordById(Long id, boolean includePassword);

    /**
     * Create a new password entry
     *
     * @param request password creation request
     * @return API response with the created password entry
     */
    ApiResponse<PasswordResponse> createPassword(PasswordCreateRequest request);

    /**
     * Update an existing password entry
     *
     * @param id password entry ID
     * @param request password update request
     * @return API response with the updated password entry
     */
    ApiResponse<PasswordResponse> updatePassword(Long id, PasswordUpdateRequest request);

    /**
     * Delete a password entry
     *
     * @param id password entry ID
     * @return API response indicating success
     */
    ApiResponse<?> deletePassword(Long id);

    /**
     * Generate a random password
     *
     * @param request password generation request
     * @return API response with the generated password
     */
    ApiResponse<GeneratePasswordResponse> generatePassword(GeneratePasswordRequest request);

    /**
     * Check the strength of a password
     *
     * @param request password strength check request
     * @return API response with the password strength
     */
    ApiResponse<CheckPasswordStrengthResponse> checkPasswordStrength(CheckPasswordStrengthRequest request);

    /**
     * Export passwords
     *
     * @param format export format (CSV, JSON)
     * @param includePasswords whether to include decrypted passwords
     * @return API response with the exported passwords
     */
    ApiResponse<ExportPasswordsResponse> exportPasswords(String format, boolean includePasswords);

    /**
     * Import passwords
     *
     * @param request password import request
     * @return API response with import results
     */
    ApiResponse<ImportPasswordsResponse> importPasswords(ImportPasswordsRequest request);
}

File: auth-client/src/services/password-manager.service.ts
import apiWrapper from './apiWrapper'
import { handleApiError } from '@/utils/errorHandler'
import { PagedResponse } from '@/types/api'

// Password interfaces
export interface Password {
  id: number
  siteUrl: string
  username: string
  password?: string
  iconUrl?: string
  createdAt: string
  updatedAt: string
}

export interface PasswordCreateRequest {
  siteUrl: string
  username: string
  password: string
  iconUrl?: string
}

export interface PasswordUpdateRequest {
  siteUrl?: string
  username?: string
  password?: string
  iconUrl?: string
}

export interface PasswordSearchRequest {
  keyword?: string
  provider?: string
  username?: string
  page?: number
  size?: number
  sortBy?: string
  sortDirection?: 'ASC' | 'DESC'
}

export interface GeneratePasswordRequest {
  length?: number
  includeLowercase?: boolean
  includeUppercase?: boolean
  includeNumbers?: boolean
  includeSpecial?: boolean
}

export interface CheckPasswordStrengthRequest {
  password: string
}

export interface CheckPasswordStrengthResponse {
  score: number
  feedback: string
}

export interface ImportPasswordsRequest {
  format: 'JSON' | 'CSV'
  importData?: string
  passwords?: PasswordCreateRequest[]
}

export interface ImportPasswordsResponse {
  totalImported: number
  failedImports: number
}

export interface ExportPasswordsResponse {
  exportData: string
}

class PasswordManagerService {
  async getPassword(id: number, includePassword: boolean = false): Promise<Password> {
    try {
      return await apiWrapper.post<Password>(`/api/passwords/get?id=${id}&includePassword=${includePassword}`)
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async searchPasswords(request: PasswordSearchRequest): Promise<PagedResponse<Password>> {
    try {
      return await apiWrapper.post<PagedResponse<Password>>('/api/passwords/search', request)
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async createPassword(request: PasswordCreateRequest): Promise<Password> {
    try {
      return await apiWrapper.post<Password>('/api/passwords/create', request)
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async updatePassword(id: number, request: PasswordUpdateRequest): Promise<Password> {
    try {
      return await apiWrapper.post<Password>(`/api/passwords/update?id=${id}`, request)
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async deletePassword(id: number): Promise<void> {
    try {
      await apiWrapper.post<void>(`/api/passwords/delete?id=${id}`)
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async generatePassword(request: GeneratePasswordRequest): Promise<string> {
    try {
      const response = await apiWrapper.post<GeneratePasswordResponse>('/api/passwords/generate', request)
      return response.password
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async checkPasswordStrength(password: string): Promise<CheckPasswordStrengthResponse> {
    try {
      return await apiWrapper.post<CheckPasswordStrengthResponse>('/api/passwords/check-strength', { password })
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async exportPasswords(format: 'JSON' | 'CSV', includePasswords: boolean = false): Promise<string> {
    try {
      const response = await apiWrapper.post<ExportPasswordsResponse>(
        `/api/passwords/export?format=${format}&includePasswords=${includePasswords}`
      )
      return response.exportData
    } catch (error) {
      throw handleApiError(error)
    }
  }

  async importPasswords(request: ImportPasswordsRequest): Promise<ImportPasswordsResponse> {
    try {
      return await apiWrapper.post<ImportPasswordsResponse>('/api/passwords/import', request)
    } catch (error) {
      throw handleApiError(error)
    }
  }
}

// Add missing interface
interface GeneratePasswordResponse {
  password: string
}

export default new PasswordManagerService()
